zrml/asset-router/src/tests/inspect.rs

```diff
@@ -0,0 +1,99 @@
+    assert!(AssetRouter::asset_exists(asset));
+    assert_ok!(<AssetRouter as orml_traits::MultiCurrency<AccountId>>::deposit(
+        asset,
+        &ALICE,
+        initial_amount
+    ));
+    assert_eq!(AssetRouter::total_issuance(asset), initial_amount);
+    assert_eq!(AssetRouter::balance(asset, &ALICE), initial_amount);
+    assert_eq!(AssetRouter::reducible_balance(asset, &ALICE, false), initial_amount);
+    assert_eq!(
+        AssetRouter::can_withdraw(asset, &ALICE, initial_amount),
+        WithdrawConsequence::ReducedToZero(0)
+    );
+    assert_eq!(AssetRouter::can_deposit(asset, &ALICE, 1, true), DepositConsequence::Success);
+}
+
+#[test]
+fn inspect_routes_campaign_assets_correctly() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(AssetRouter::create(CAMPAIGN_ASSET, ALICE, true, CAMPAIGN_ASSET_MIN_BALANCE,));
+
+        inspect_test_helper(CAMPAIGN_ASSET, CAMPAIGN_ASSET_INITIAL_AMOUNT);
+
+        assert_eq!(AssetRouter::total_issuance(CUSTOM_ASSET), 0);
+        assert_eq!(AssetRouter::total_issuance(MARKET_ASSET), 0);
+        assert_eq!(AssetRouter::total_issuance(CURRENCY), 0);
+    });
+}
+
+#[test]
+fn inspect_routes_custom_assets_correctly() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(AssetRouter::create(CUSTOM_ASSET, ALICE, true, CUSTOM_ASSET_MIN_BALANCE,));
+
+        inspect_test_helper(CUSTOM_ASSET, CUSTOM_ASSET_INITIAL_AMOUNT);
+
+        assert_eq!(AssetRouter::total_issuance(CAMPAIGN_ASSET), 0);
+        assert_eq!(AssetRouter::total_issuance(MARKET_ASSET), 0);
+        assert_eq!(AssetRouter::total_issuance(CURRENCY), 0);
+    });
+}
+
+#[test]
+fn inspect_routes_market_assets_correctly() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(AssetRouter::create(MARKET_ASSET, ALICE, true, MARKET_ASSET_MIN_BALANCE,));
+
+        inspect_test_helper(MARKET_ASSET, MARKET_ASSET_INITIAL_AMOUNT);
+
+        assert_eq!(AssetRouter::total_issuance(CAMPAIGN_ASSET), 0);
+        assert_eq!(AssetRouter::total_issuance(CUSTOM_ASSET), 0);
+        assert_eq!(AssetRouter::total_issuance(CURRENCY), 0);
+    });
+}
+
+#[test]
+fn multicurrency_routes_currencies_correctly() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(<AssetRouter as orml_traits::MultiCurrency<AccountId>>::deposit(
+            CURRENCY,
+            &ALICE,
+            CURRENCY_INITIAL_AMOUNT
+        ));
+        assert_eq!(AssetRouter::total_issuance(CURRENCY), 0);
+        assert_eq!(AssetRouter::balance(CURRENCY, &ALICE), 0);
+        assert_eq!(AssetRouter::reducible_balance(CURRENCY, &ALICE, false), 0);
+        assert_eq!(
+            AssetRouter::can_withdraw(CURRENCY, &ALICE, CURRENCY_INITIAL_AMOUNT),
+            WithdrawConsequence::UnknownAsset
+        );
+        assert_eq!(
+            AssetRouter::can_deposit(CURRENCY, &ALICE, 1, true),
+            DepositConsequence::UnknownAsset
+        );
+    });
```

Comment:
I would argue that this is not correct behavior. If all functions would throw, this would be fine, but with `total_issuance` returning zero, it feels a bit odd. What's the rationale for not implementing `Inspect` by routing to `MultiCurrency` when you get a `Currency` instance?

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1189/commits/cf6af0770a1e666f956c994db2fab0a0d0d9d8a4